pkg/kubectl/cmd/util/printing.go

```diff
@@ -223,3 +223,17 @@ func ValidResourceTypeList(f ClientAccessFactory) string {
 	
 	`)
 }
+
+// Retrieve a list of handled resources from printer as valid args
```

Comment:
keep the `TODO: this needs to be abstracted behind the factory like ValidResourceTypeList and use discovery`... this change removes duplicate code, but using a statically compiled-in list is still wrong

Reply:
thanks @liggitt , will add the TODO back soon!